src/Time/ChangeSlabSize/Projector.hpp

```diff
@@ -0,0 +1,67 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <map>
+#include <unordered_map>
+#include <utility>
+
+#include "Domain/Structure/Element.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "ParallelAlgorithms/Amr/Protocols/Projector.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace Tags::ChangeSlabSize {
+struct NumberOfExpectedMessages;
+}  // namespace Tags::ChangeSlabSize
+/// \endcond
+
+/// Project data relating to the ChangeSlabSize action during AMR.
+struct ProjectChangeSlabSize : tt::ConformsTo<amr::protocols::Projector> {
+  using return_tags =
+      tmpl::list<::Tags::ChangeSlabSize::NumberOfExpectedMessages>;
+  using argument_tags = tmpl::list<>;
+
+  template <size_t Dim>
+  static void apply(
+      const gsl::not_null<std::map<int64_t, size_t>*>
+      /*number_of_expected_messages*/,
+      const std::pair<Mesh<Dim>, Element<Dim>>& /*old_mesh_and_element*/) {
+    // Do not change anything for p-refinement
+  }
+
+  template <typename... Tags>
+  static void apply(const gsl::not_null<std::map<int64_t, size_t>*>
+                        number_of_expected_messages,
+                    const tuples::TaggedTuple<Tags...>& parent_items) {
+    *number_of_expected_messages =
+        tuples::get<::Tags::ChangeSlabSize::NumberOfExpectedMessages>(
+            parent_items);
+  }
+
+  template <size_t Dim, typename... Tags>
+  static void apply(
+      const gsl::not_null<std::map<int64_t, size_t>*>
+          number_of_expected_messages,
+      const std::unordered_map<ElementId<Dim>, tuples::TaggedTuple<Tags...>>&
+          children_items) {
+    *number_of_expected_messages =
+        tuples::get<::Tags::ChangeSlabSize::NumberOfExpectedMessages>(
+            children_items.begin()->second);
+    for (const auto& child : children_items) {
+      ASSERT(*number_of_expected_messages ==
+                 tuples::get<::Tags::ChangeSlabSize::NumberOfExpectedMessages>(
+                     child.second),
+             "Different elements expecting different numbers of "
+             "ChangeSlabSize messages.");
+    }
+  }
+};
```

Comment:
I believe this projector is currently the same as the generic projector `ParallelAlgorithms/Amr/Projectors/CopyFromCreatorOrLeaveAsIs.hpp` 
I leave it up to you as to whether you want to keep this or use the generic projector...